pkg/client/clientcmd/overrides.go

```diff
@@ -141,7 +142,7 @@ func BindAuthInfoFlags(authInfo *clientcmdapi.AuthInfo, flags *pflag.FlagSet, fl
 // BindClusterFlags is a convenience method to bind the specified flags to their associated variables
 func BindClusterFlags(clusterInfo *clientcmdapi.Cluster, flags *pflag.FlagSet, flagNames ClusterOverrideFlags) {
 	flags.StringVarP(&clusterInfo.Server, flagNames.APIServer, flagNames.APIServerShort, "", "The address and port of the Kubernetes API server")
-	flags.StringVar(&clusterInfo.APIVersion, flagNames.APIVersion, "", "The API version to use when talking to the server")
+	flags.StringVar(&clusterInfo.APIVersion, flagNames.APIVersion, latest.Version, "The API version to use when talking to the server")
```

Comment:
This means you're leaving dead defaulting code all over. If we were to require the version to be specified via the flag (including flag default value) and/or in API objects (for create/update), then we should just require that and eliminate other defaults.

For example:
https://github.com/GoogleCloudPlatform/kubernetes/blob/b6f6de150afe566da8305e224d776d6dc7a6ce6e/pkg/kubectl/cmd/cmd.go#L344

Unfortunately, the means by which the version is set, from latest, mapping.APIVersion, flag, and/or config is confusing and spread all over.

You don't have to solve that problem now, though -- maybe revert and file an issue to come back to it.